vms/platformvm/state/state_test.go

```diff
@@ -2088,3 +2088,257 @@ func TestSubnetOnlyValidatorAfterLegacyRemoval(t *testing.T) {
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			require := require.New(t)
+
+			db := memdb.New()
+			state := newTestState(t, db)
+
+			for _, staker := range test.initial {
+				primaryStaker := &Staker{
+					TxID:      ids.GenerateTestID(),
+					SubnetID:  constants.PrimaryNetworkID,
+					NodeID:    staker.NodeID,
+					PublicKey: staker.PublicKey,
+					Weight:    5,
+					StartTime: staker.StartTime.Add(-1 * time.Second),
+				}
+				require.NoError(state.PutCurrentValidator(primaryStaker))
+				require.NoError(state.PutCurrentValidator(staker))
+			}
+
+			for _, sov := range test.sovs {
+				sov.RemainingBalanceOwner = []byte{}
+				sov.DeactivationOwner = []byte{}
+
+				require.NoError(state.PutSubnetOnlyValidator(sov))
+			}
+
+			state.SetHeight(0)
+			require.NoError(state.Commit())
+
+			stakersBySubnetID := make(map[ids.ID][]*Staker)
+			for _, staker := range test.initial {
+				stakers := stakersBySubnetID[staker.SubnetID]
+				stakersBySubnetID[staker.SubnetID] = append(stakers, staker)
+			}
+
+			sovBySubnetID := make(map[ids.ID][]SubnetOnlyValidator)
+			for _, sov := range test.sovs {
+				if sov.Weight == 0 {
+					continue
+				}
+				sovs := sovBySubnetID[sov.SubnetID]
+				sovBySubnetID[sov.SubnetID] = append(sovs, sov)
+			}
+
+			for _, subnetID := range subnetIDs {
+				currentValidators, height, err := state.GetCurrentValidatorSet(context.Background(), subnetID)
+				require.NoError(err)
+				require.Equal(uint64(0), height)
+				totalLen := len(stakersBySubnetID[subnetID]) + len(sovBySubnetID[subnetID])
+				require.Len(currentValidators, totalLen)
+
+				for _, expectedStaker := range stakersBySubnetID[subnetID] {
+					currentValidator, ok := currentValidators[expectedStaker.TxID]
+					require.True(ok)
+					require.Equal(expectedStaker.TxID, currentValidator.ValidationID)
+					require.Equal(expectedStaker.NodeID, currentValidator.NodeID)
+					require.Equal(expectedStaker.PublicKey, currentValidator.PublicKey)
+					require.Equal(expectedStaker.Weight, currentValidator.Weight)
+					require.Equal(uint64(expectedStaker.StartTime.Unix()), currentValidator.StartTime)
+					require.Equal(uint64(0), currentValidator.SetWeightNonce)
+					require.True(currentValidator.IsActive)
+					require.False(currentValidator.IsSoV)
+				}
+
+				for _, expectedSOV := range sovBySubnetID[subnetID] {
+					currentValidator, ok := currentValidators[expectedSOV.ValidationID]
+					require.True(ok)
+					require.Equal(expectedSOV.ValidationID, currentValidator.ValidationID)
+					require.Equal(expectedSOV.NodeID, currentValidator.NodeID)
+					require.Equal(expectedSOV.PublicKey, currentValidator.PublicKey.Serialize())
+					require.Equal(expectedSOV.Weight, currentValidator.Weight)
+					require.Equal(expectedSOV.StartTime, currentValidator.StartTime)
+					require.Equal(expectedSOV.MinNonce, currentValidator.SetWeightNonce)
+					require.Equal(expectedSOV.isActive(), currentValidator.IsActive)
```

Comment:
should there be some inputs with `EndAccumulatedFee > 0` so sometimes .isActive returns true?

Reply:
yea good idea, added few